hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Resource.java

```diff
@@ -16,140 +16,21 @@
-     * called.
-     */
-    @ApiModel("ResourceBlueprint")
-    public static final class Blueprint extends Entity.Blueprint {
-        private final String resourceTypePath;
-
-        public static Builder builder() {
-            return new Builder();
-        }
-
-        /**
-         * JAXB support
-         */
-        @SuppressWarnings("unused")
-        private Blueprint() {
-            resourceTypePath = null;
-        }
-
-        public Blueprint(String id, String resourceTypePath) {
-            this(id, resourceTypePath, Collections.emptyMap());
-        }
-
-        public Blueprint(String id, String resourceTypePath, Map<String, Object> properties) {
-            super(id, properties);
-            this.resourceTypePath = resourceTypePath;
-        }
-
-        public Blueprint(String id, String resourceTypePath, Map<String, Object> properties,
-                         Map<String, Set<CanonicalPath>> outgoing,
-                         Map<String, Set<CanonicalPath>> incoming) {
-            super(id, properties, outgoing, incoming);
-            this.resourceTypePath = resourceTypePath;
-        }
-
-        public Blueprint(String id, String name, String resourceTypePath, Map<String, Object> properties,
-                         Map<String, Set<CanonicalPath>> outgoing,
-                         Map<String, Set<CanonicalPath>> incoming) {
-            super(id, name, properties, outgoing, incoming);
-            this.resourceTypePath = resourceTypePath;
-        }
-
-        public String getResourceTypePath() {
-            return resourceTypePath;
-        }
-
-        @Override
-        public <R, P> R accept(ElementBlueprintVisitor<R, P> visitor, P parameter) {
-            return visitor.visitResource(this, parameter);
-        }
-
-        public static final class Builder extends Entity.Blueprint.Builder<Blueprint, Builder> {
-            private String resourceTypePath;
-
-            public Builder withResourceTypePath(String resourceTypePath) {
-                this.resourceTypePath = resourceTypePath;
-                return this;
-            }
-
-            @Override
-            public Blueprint build() {
-                return new Blueprint(id, name, resourceTypePath, properties, outgoing, incoming);
-            }
-        }
-    }
+public class Resource {
+    private String id;  // Unique index [Search resource by id]
+    private String name;
+    private ResourceType type;  // Index [Search all resources of type xx]
+    private String feed;    // Index; But not sure if feeds are still in play if the inventory is built from directly prometheus scans
+    private String rootId;  // Nullable; Index [Search all resources under root xx]
+    private Map<String, String> properties;
+
+    // Maybe "children" and "metrics" should be removed from "Resource" and put in another class that represents the whole tree
+    private Collection<Resource> children;
```

Comment:
Perhaps this can be modelled in a lazy way.
The design is good, but probably I will do this in two steps:
a) Give me Resource id1
b) Give me children of id1
So, I wonder if when we return a resource, we need to collect all children + metrics or that can be on demand.
(Or we can define this thinking on the flows we need to support).
Same with metrics, perhaps I want the metrics of a resource but I'm not interested in the children.

Reply:
But yes, design is ok. Just thinking loud.
(My comments are not a formal revision, just catching up or making suggestions with all the feedback we are collecting)

Reply:
+1 for lazy loading